src/scripts.js

```diff
@@ -76,15 +80,48 @@ function getDirections(event){
 };
 
 
+function searchByName(){
+  let recipeRepo = new RecipeRepository(recipeData);
+  let filteredRecipes = recipeRepo.filterByName(searchFieldInput.value)
+  show(allRecipeGrid);
+  hide(recipeGrid);
+
+  const recipeCard = filteredRecipes.reduce((acc, recipe) => {
+    acc +=
+      `<article class="mini-recipe" id="${recipe.id}">
+       <img src= "${recipe.image}" alt= "${recipe.name}">
+       <p>${recipe.name}</p>
+       </article>`
+
+    return acc;
+  }, []);
+  return allRecipeGrid.innerHTML = recipeCard
+
+}
+
+
+
+
 function searchData(input) {
-  let searchedData = recipeData.filter(recipe => recipe[`${input}`].includes(searchBar.input));
+  let searchedData = recipeData.filter(recipe => recipe[`${input}`].includes(searchFieldInput.value));
+  return searchedData
```

Comment:
I'm not sure we need this helper function anymore. We can see if it's helpful for tags, but if not, let's remember to delete it